docs/src/content/docs/guides/sidebar.mdx

```diff
@@ -0,0 +1,382 @@
+---
+title: Sidebar Customization
+description: Learn how to customize the sidebar of your documentation.
+---
+
+import FileTree from '../../../components/file-tree.astro';
+import SidebarPreview from '../../../components/sidebar-preview.astro';
+
+A well-organized sidebar is key to a good documentation as it is one of the main ways users will navigate your site. Starlight provides a complete set of options to customize your sidebar layout and content.
+
+## Default sidebar
+
+By default, in the absence of a custom sidebar configuration, Starlight will automatically generate a sidebar based on the filesystem structure of your documentation.
+
+For example, given the following file structure:
+
+<FileTree>
+
+- src/
+  - content/
+    - docs/
+      - guides/
+        - components.md
+        - i18n.md
+      - reference/
+        - configuration.md
+
+</FileTree>
+
+The following sidebar will be automatically generated:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'guides',
+			items: [
+				{ label: 'Components', link: '/guides/components/' },
+				{ label: 'Internationalization (i18n)', link: '/guides/i18n/' },
+			],
+		},
+		{
+			label: 'reference',
+			items: [
+				{ label: 'Configuration Reference', link: '/reference/configuration/' },
+			],
+		},
+	]}
+/>
+
+Each link label will by default match the [`title` frontmatter field](/reference/configuration/#title-required) required on each page.
+
+Learn more about autogenerated sidebars in the [autogenerated groups](#autogenerated-groups) section.
+
+## Basic configuration
+
+The main configuration for the sidebar is located in the [`sidebar` field](/reference/configuration/#sidebar) of your Starlight configuration located in the `astro.config.mjs` file.
+
+A sidebar is primarily composed of two elements:
+
+- [Links](#links)
+- [Groups of links](#groups)
+
+Configuring your sidebar is as simple a defining an array of links and groups in the `sidebar` field of your Starlight configuration.
+
+### Links
+
+A link to an internal or external page must be defined by a `label` and a `link` property.
+
+```js
+starlight({
+	sidebar: [
+		// A link to the CSS & Styling guide.
+		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/' },
+		// An external link to the Astro website.
+		{ label: 'Astro', link: 'https://astro.build/' },
+	],
+});
+```
+
+With the above configuration, the following sidebar will be generated:
+
+<SidebarPreview
+	config={[
+		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/' },
+		{ label: 'Astro', link: 'https://astro.build/' },
+	]}
+/>
+
```

Comment:
🏷️ guide
````suggestion
Optionally, a link can include a `badge` property to display a badge next to the link label.

```js
starlight({
	sidebar: [
		// A link to the CSS & Styling guide.
		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/', badge: 'New' },
		// An external link to the Astro website.
		{ label: 'Astro', link: 'https://astro.build/' },
	],
});
```

The above configuration will generate the following sidebar:

<SidebarPreview
	config={[
		{ label: 'CSS & Styling', link: '/guides/css-and-tailwind/', badge: 'New' },
		{ label: 'Astro', link: 'https://astro.build/' },
	]}
/>

To further customize the appearance of the badge you can pass an object to the `badge` property containing a `text` and a `variant` property. The `variant` property can be set to `blue`, `green`, `red`, `pink` or `yellow`. By default, the `variant` property is set to `default`, which will uses the accent color of your site.

```js
starlight({
	sidebar: [
		// A link to the CSS & Styling guide.
		{
			label: 'CSS & Styling',
			link: '/guides/css-and-tailwind/',
			badge: { text: 'New', variant: 'green' },
		},
		// An external link to the Astro website.
		{ label: 'Astro', link: 'https://astro.build/' },
	],
});
```

The above configuration will generate the following sidebar:

<SidebarPreview
	config={[
		{
			label: 'CSS & Styling',
			link: '/guides/css-and-tailwind/',
			badge: { text: 'New', variant: 'green' },
		},
		{ label: 'Astro', link: 'https://astro.build/' },
	]}
/>

````

Reply:
Thanks a lot @kevinzunigacuellar. I made a few changes in my initial PR to ensure badges previews render properly and added your suggestions with a few changes (do not hesitate to tell me if you disagree with any of them):

- I moved the badges part of the guide to their own section. I think they deserve it and it also adds a ToC entry for them so easier to quickly jump/link to them.
- I added an example of the frontmatter badge configuration for autogenerated link items.

It currently looks like this:

![badges](https://github.com/withastro/starlight/assets/494699/732fdc03-4139-4283-b8c6-699247136dc6)

Reply:
I like it 💜